website/project/model.py

```diff
@@ -2444,21 +2444,26 @@ def register_node(self, schema, auth, data, parent=None):
         # After register callback
         for addon in original.get_addons():
             _, message = addon.after_register(original, registered, auth.user)
-            if message:
+            if message and not recur:
                 status.push_status_message(message, kind='info', trust=False)
 
-        for node_contained in original.nodes:
-            if not node_contained.is_deleted:
-                child_registration = node_contained.register_node(
-                    schema=schema,
-                    auth=auth,
-                    data=data,
-                    parent=registered,
-                )
-                if child_registration and not child_registration.primary:
-                    registered.nodes.append(child_registration)
+        children_nodes = original.nodes
+        while(children_nodes):
+            batch = children_nodes[:30]
+            for node_contained in batch:
+                if not node_contained.is_deleted:
+                    child_registration = node_contained.register_node(
+                        schema=schema,
+                        auth=auth,
+                        data=data,
+                        parent=registered,
+                        recur=True,
+                    )
+                    if child_registration and not child_registration.primary:
```

Comment:
If this is the check to see if this is a node pointer, that's different now in DjangOSF. There is, I think, a better way to get the node hierarchy in the node model at this point. To verify, make a pointer from a child to a parent, from a parent to a different child, and a child to itself, then see if this works or if it just goes forever / hits the maximum recursion depth.

This might not be the right time to try to eliminate this recursive function, but at least verify that it doesn't break with node pointers as described above.